src/promptflow/promptflow/_sdk/data/executable/flow_test_main.py.jinja2

```diff
@@ -0,0 +1,107 @@
+
+    def render_message(role, message_items):
+        with st.chat_message(role):
+            dict_iter_render_message(message_items)
+
+    def show_conversation() -> None:
+        if "messages" not in st.session_state:
+            st.session_state.messages = []
+            st.session_state.history = []
+        if st.session_state.messages:
+            for role, message_items in st.session_state.messages:
+                render_message(role, message_items)
+
+
+    def get_chat_history_from_session():
+        if "history" in st.session_state:
+            return st.session_state.history
+        return []
+
+
+    def submit(**kwargs) -> None:
+        st.session_state.messages.append(("user", kwargs))
+        session_state_history = dict()
+        session_state_history.update({"inputs": kwargs})
+        with container:
+            render_message("user", kwargs)
+        # Append chat history to kwargs
+        response = run_flow({'{{chat_history_input_name}}': get_chat_history_from_session(), **kwargs})
+        st.session_state.messages.append(("assistant", response))
+        session_state_history.update({"outputs": response})
+        st.session_state.history.append(session_state_history)
+        with container:
+            render_message("assistant", response)
+
+
+    def run_flow(data: dict) -> dict:
+        global invoker
+        if not invoker:
+            flow = Path('{{flow_path}}')
+            dump_path = Path('{{flow_path}}').parent
+            if flow.is_dir():
+                os.chdir(flow)
+            else:
+                os.chdir(flow.parent)
+            invoker = FlowInvoker(flow, connection_provider="""{{ connection_provider }}""", dump_to=dump_path)
+        result = invoker.invoke(data)
+        return result
+
+    image = Image.open(Path(__file__).parent / "logo.png")
+    st.set_page_config(
+        layout="wide",
+        page_title="{{flow_name}} - Promptflow App",
+        page_icon=image,
+        menu_items={
+            'About': """
+            # This is a Promptflow App.
+
+            You can refer to [promptflow](https://github.com/microsoft/promptflow) for more information.
+            """
+        }
+    )
+    # Set primary button color here since button color of the same form need to be identical in streamlit, but we only need Run/Chat button to be blue.
+    st.config.set_option("theme.primaryColor", "#0F6CBD")
+    st.title("{{flow_name}}")
+    st.divider()
+    st.chat_message("assistant").write("Hello, please input following flow inputs.")
+    container = st.container()
+    with container:
+        show_conversation()
+
+    with st.form(key='input_form', clear_on_submit=True):
+        st.text('{{chat_input_name}}')
+        {{chat_input_name}} = st_quill(html=True, toolbar=["image"], key='{{chat_input_name}}', placeholder='Please enter the list values and use the image icon to upload a picture. Make sure to format each list item correctly with line breaks')
```

Comment:
Are all inputs list formats in chat_flow?

Reply:
Yes, there is only one input is allowed in chat flow, and also one output.

Reply:
and the one input should always be type list?

Reply:
where do we do the check?

Reply:
At StreamlitFileGenerator(BaseGenerator) init, _is_chat_flow()